src/submodule.c

```diff
@@ -1916,31 +1917,33 @@ static int submodule_load_from_wd_lite(git_submodule *sm)
 
 /**
  * Returns a snapshot of $WORK_TREE/.gitmodules.
- *
- * We ignore any errors and just pretend the file isn't there.
  */
-static git_config *gitmodules_snapshot(git_repository *repo)
+static int gitmodules_snapshot(git_repository *repo, git_config **snap)
 {
 	const char *workdir = git_repository_workdir(repo);
-	git_config *mods = NULL, *snap = NULL;
-	git_buf path = GIT_BUF_INIT;
+
+	*snap = NULL;
 
 	if (workdir != NULL) {
+		git_buf path = GIT_BUF_INIT;
+		git_config *mods = NULL;
 		if (git_buf_joinpath(&path, workdir, GIT_MODULES_FILE) != 0)
-			return NULL;
+			return 0;
 
-		if (git_config_open_ondisk(&mods, path.ptr) < 0)
-			mods = NULL;
-	}
+		int error = git_config_open_ondisk(&mods, path.ptr);
```

Comment:
Might be I'm misreading the diff, but isn't this introducing a declaration after instructions?